polkadot/xcm/src/v5/traits.rs

```diff
@@ -289,7 +295,8 @@ pub trait PreparedMessage {
 /// Type of XCM message executor.
 pub trait ExecuteXcm<Call> {
 	type Prepared: PreparedMessage;
-	fn prepare(message: Xcm<Call>) -> result::Result<Self::Prepared, Xcm<Call>>;
+	/// If it fails, returns the index of the problematic instruction.
+	fn prepare(message: Xcm<Call>) -> result::Result<Self::Prepared, u8>;
```

Comment:
And the last thing, what I am thinking about, would it make sense to change here `u8` to `OutcomeError`?
```suggestion
	fn prepare(message: Xcm<Call>) -> result::Result<Self::Prepared, OutcomeError>;
```

Maybe the `OutcomeError` is not the best name and it could be more like `InstructionError`